### ISSUE TITLE
Fallback locale does not fork

### ISSUE THREAD

**Issue by akolpakov** [joshswan/react-native-globalize#59]
In case if we use locale that does not presented. For example:

```
loadMessages({
  'en': {
    welcome: 'Hallo, heute ist der {date}',
  }
})
```

And we use `GlobalizeProvider` with `localeFallback: "en"`

When user is in locale `sv` for example, we get the following error:

```
[RNGlobalize] CLDR data for the selected language/locale has not been loaded!
```

The reason, that in the code we have:

https://github.com/joshswan/react-native-globalize/blob/3e3c63a4047d38711dd633ec1712f16c4092014c/src/globalize/index.ts#L49

It means if cfg.fallback is "en", we still try to `findFallbackLocale` for locale `sv` which does not exist

We need to change it to:

```
cfg.locale = (cfg.fallback && findFallbackLocale(cfg.fallback)) as string;
```

**Comment by joshswan** [comment]
As documented in the types and the docs, `localeFallback` and `cfg.fallback` are meant to be booleans currently, simply enabling or disabling the auto-fallback behavior (e.g. en-US -> en). Changing the implementation like this would break existing code and require a major version bump. I think a separate config prop that's opt-in would be a better solution.

**Comment by akolpakov** [comment]
I got you.

But how to avoid the case when user's locale does not exists in the loadMessages?

**Comment by joshswan** [comment]
Yeah I think this is a valid enhancement, just need to figure out the implementation details. Looking into implementing a new config prop `defaultLocale` as we speak. Will update shortly.

**Comment by joshswan** [comment]
Released v4.4.0. Setting `defaultLocale="en"` on `GlobalizeProvider` should now behave as described above (i.e. it won't throw in the event of an unknown locale and will instead use the en locale).
